pkg/expression/scalar_function.go

```diff
@@ -141,9 +141,27 @@ func (sf *ScalarFunction) String() string {
 	return buffer.String()
 }
 
-// MarshalJSON implements json.Marshaler interface.
-func (sf *ScalarFunction) MarshalJSON() ([]byte, error) {
```

Comment:
Why remove this?

Reply:
It's useless. I removed some implementation of `MarshalJSON` in https://github.com/pingcap/tidb/pull/53922, but I forgot to remove the implementation of `ScalarFunction` in that PR :facepalm: .

It's originally introduced to implement `explain` (a long time ago), but it's not used anymore now.